### PR TITLE
fix(download): show appropriate download error

### PR DESCRIPTION
**Pull request by priyajeet**
for when download might get disabled due to an access policy while the user is already previewing the file without browser refresh.

Also fixes the apostrophes

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: src/lib/Preview.js [hunk 1]
```diff
@@ -513,25 +513,27 @@ class Preview extends EventEmitter {
...
      * @public
-     * @return {void}
+     * @return {Promise} download promise
      */
     download() {
         const downloadErrorMsg = __('notification_cannot_download');
         if (!canDownload(this.file, this.options)) {
             this.ui.showNotification(downloadErrorMsg);
-            return;
+            return Promise.reject();
```
Comment (ConradJChan) [inline src/lib/Preview.js]: Should we reject with a reason?

File: src/lib/Preview.js [hunk 2]
```diff
@@ -513,25 +513,27 @@ class Preview extends EventEmitter {
      * Downloads the file being previewed.
      *
      * @public
-     * @return {void}
+     * @return {Promise} download promise
```
Comment (ConradJChan) [inline src/lib/Preview.js]: This is a public API for the preview instance, right? Is this considered a breaking change?

File: src/lib/Preview.js [hunk 3]
```diff
@@ -541,15 +543,25 @@ class Preview extends EventEmitter {
...
             const getDownloadUrl = appendQueryParams(getDownloadURL(this.file.id, apiHost), queryParams);
-            this.api.get(getDownloadUrl, { headers: this.getRequestHeaders() }).then(data => {
-                const downloadUrl = appendQueryParams(data.download_url, queryParams);
-                this.api.reachability.downloadWithReachabilityCheck(downloadUrl);
-            });
+            downloadPromise = this.api
+                .get(getDownloadUrl, { headers: this.getRequestHeaders() })
+                .then(data => {
+                    const downloadUrl = appendQueryParams(data.download_url, queryParams);
+                    this.api.reachability.downloadWithReachabilityCheck(downloadUrl);
```
Comment (ConradJChan) [inline src/lib/Preview.js]: Should this return the promise result from `downloadWithReachabilityCheck`?

File: src/lib/Preview.js [hunk 4]
```diff
@@ -541,15 +543,25 @@ class Preview extends EventEmitter {
...
+                })
+                .catch(error => {
+                    const code = getProp(error, 'response.data.code');
+                    let msg = downloadErrorMsg;
+                    if (code === 'forbidden_by_policy') {
+                        msg = __('notification_cannot_download_due_to_policy');
+                    }
+                    this.ui.showNotification(msg);
```
Comment (ConradJChan) [inline src/lib/Preview.js]: Should we do a `Promise.reject` here?

File: src/lib/Preview.js [hunk 5]
```diff
@@ -541,15 +543,25 @@ class Preview extends EventEmitter {
                 params,
             );
 
-            this.api.reachability.downloadWithReachabilityCheck(downloadUrl);
+            downloadPromise = this.api.reachability.downloadWithReachabilityCheck(downloadUrl);
```
Comment (ConradJChan) [inline src/lib/Preview.js]: It doesn't look like `downloadWithReachabilityCheck` actually returns a promise?

File: src/i18n/en-AU.properties [hunk 6]
```diff
@@ -41,11 +41,11 @@ has_x_refs=This preview has references you cannot view. Open the file in its nat
 
 # Error messages
 # Default preview error message
-error_generic=We're sorry, the preview didn't load.
+error_generic=We’re sorry, the preview didn't load.
```
Comment (ConradJChan) [inline src/i18n/en-AU.properties]: What was wrong with the apostrophes?

File: src/lib/Preview.js (hunk 2 above)
Comment (priyajeet) [reply src/lib/Preview.js]: I am actually gonna remove the return promise. Only downside is that this function is not really being tested properly, specially for errors.

File: src/i18n/en-AU.properties (hunk 6 above)
Comment (priyajeet) [reply src/i18n/en-AU.properties]: Normally for user facing strings one uses the accented versions. It also helps with not having the need to escape.

**Pull request merged**